src/server/session.ts

```diff
@@ -21,6 +21,21 @@ namespace ts.server {
         return spaceCache[n];
     }
 
+    export function generateIndentString(n: number, editorOptions: EditorOptions): string {

```

Comment:
we already have similar function [getIndentationString](https://github.com/Microsoft/TypeScript/blob/master/src/services/formatting/formatting.ts#L1071). Can it be used instead to avoid code duplication?